src/websockets/binary_websockets.es6

```diff
@@ -149,14 +152,30 @@ const needs_authentication = (data) => {
    return false;
 };
 
+const website_status = (data) => {
```

Comment:
@arnabk there is a flag that is used to stop sending any WebSocket requests and buffer them. Instead of moving it to the navigation folder I made a separate folder to hold the logic for this. 

There are a couple of lines of code which deals with checking the flag `is_website_up` and sending all the buffered requests.